### PR TITLE
Use gz_vendor packages

### PR DESCRIPTION
**Pull request by azeey**
# 🎉 New feature

## Summary
Changes Gazebo dependencies to use vendor packages (see https://github.com/gazebo-tooling/release-tools/issues/1117)

## Test it
Build and run tests.

## Checklist
- [ ] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

**Review by ahcorde — CHANGES_REQUESTED** [review]
are packages released and sync?
```bash
ERROR: the following rosdeps failed to install
  apt: command [apt-get install -y ros-rolling-sdformat-urdf] failed
  apt: command [apt-get install -y ros-rolling-gz-transport-vendor] failed
  apt: Failed to detect successful installation of [ros-rolling-sdformat-urdf]
  apt: Failed to detect successful installation of [ros-rolling-gz-transport-vendor]
```

**Comment by azeey** [comment]
> are packages released and sync?
> 
> ```shell
> ERROR: the following rosdeps failed to install
>   apt: command [apt-get install -y ros-rolling-sdformat-urdf] failed
>   apt: command [apt-get install -y ros-rolling-gz-transport-vendor] failed
>   apt: Failed to detect successful installation of [ros-rolling-sdformat-urdf]
>   apt: Failed to detect successful installation of [ros-rolling-gz-transport-vendor]
> ```

Not quite yet. sdformat_urdf is waiting for https://github.com/ros/sdformat_urdf/pull/28 to be merged and released. 
gz-transport-vendor is released, but it's failing to build on Noble due to an issue in Noble itself (can't install python3-distutils).

**Comment by luca-della-vedova** [comment]
I'm trying this PR with the vendored gz binaries in noble, was stuck at yaml-cpp yesterday but saw that is already fixed (thanks!), it seems now ros_gz_image is failing:

```
lucadv@noble:~/noble_ws$ colcon build --packages-select ros_gz_image
Starting >>> ros_gz_image
--- stderr: ros_gz_image                         
CMake Error at /opt/ros/rolling/opt/gz_transport_vendor/extra_cmake/lib/cmake/gz-transport/gz-transport-config.cmake:24 (add_library):
  add_library cannot create imported target "gz-transport" because another
  target with the same name already exists.
Call Stack (most recent call first):
  CMakeLists.txt:20 (find_package)


CMake Error at /opt/ros/rolling/opt/gz_msgs_vendor/extra_cmake/lib/cmake/gz-msgs/gz-msgs-config.cmake:24 (add_library):
  add_library cannot create imported target "gz-msgs" because another target
  with the same name already exists.
Call Stack (most recent call first):
  CMakeLists.txt:23 (find_package)
```

File: ros_gz_image/CMakeLists.txt [hunk 1]
```diff
@@ -16,32 +16,11 @@ find_package(ros_gz_bridge REQUIRED)
...
-# Garden
-if("$ENV{GZ_VERSION}" STREQUAL "garden")
-  find_package(gz-transport12 REQUIRED)
-  find_package(gz-msgs9 REQUIRED)
-
-  set(GZ_TARGET_PREFIX gz)
-  set(GZ_MSGS_VER ${gz-msgs9_VERSION_MAJOR})
-  set(GZ_TRANSPORT_VER ${gz-transport12_VERSION_MAJOR})
+find_package(gz_transport_vendor REQUIRED)
+find_package(gz-transport REQUIRED)
```
Comment (luca-della-vedova) [inline ros_gz_image/CMakeLists.txt:20]: It seems removing this (and `gz-msgs` below) fixes this package's build, but not sure that is the right approach.

Comment (azeey) [reply ros_gz_image/CMakeLists.txt:20]: The issue was in the upstream gz_msg vendor package. A similar fix to https://github.com/gazebo-release/gz_utils_vendor/pull/3 has been released to all the vendor packages, but hasn't been synced to ros-testing yet. Hopefully by tomorrow.

**Comment by azeey** [comment]
@ahcorde this is ready to review. CI is green 🎉

**Review by ahcorde — APPROVED** [review]

**Pull request merged**